Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/GlideUtils.java

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package org.dolphinemu.dolphinemu.utils;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.widget.ImageView;
+
+import com.bumptech.glide.Glide;
+
+import org.dolphinemu.dolphinemu.R;
+import org.dolphinemu.dolphinemu.features.settings.model.BooleanSetting;
+import org.dolphinemu.dolphinemu.model.GameFile;
+
+import java.io.File;
+
+public class GlideUtils
+{
+  public static void loadGameBanner(ImageView imageView, GameFile gameFile)
+  {
+    Context context = imageView.getContext();
+    int[] vector = gameFile.getBanner();
+    int width = gameFile.getBannerWidth();
+    int height = gameFile.getBannerHeight();
+    if (width > 0 && height > 0)
+    {
+      Bitmap bitmap = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
+      bitmap.setPixels(vector, 0, width, 0, 0, width, height);
+      Glide.with(context)
+              .load(bitmap)
+              .centerCrop()
+              .into(imageView);
+    }
+    else
+    {
+      Glide.with(context)
+              .load(R.drawable.no_banner)
+              .centerCrop()
+              .into(imageView);
+    }
+  }
+
+  public static void loadGameCover(ImageView imageView, GameFile gameFile)
+  {
+    Context context = imageView.getContext();
+    File cover = new File(gameFile.getCustomCoverPath());
+    if (cover.exists())
+    {
+      Glide.with(context)
+              .load(cover)
+              .centerCrop()
+              .error(R.drawable.no_banner)
+              .into(imageView);
+    }
+    // GameTDB has a pretty close to complete collection for US/EN covers. First pass at getting
+    // the cover will be by the disk's region, second will be the US cover, and third EN.
+    else if (BooleanSetting.MAIN_USE_GAME_COVERS.getBooleanGlobal())
+    {
+      Glide.with(context)
+              .load(CoverHelper.buildGameTDBUrl(gameFile, CoverHelper.getRegion(gameFile)))
+              .centerCrop()
+              .error(
+                      Glide.with(context)
+                              .load(CoverHelper.buildGameTDBUrl(gameFile, "US"))
+                              .centerCrop()
+                              .error(
+                                      Glide.with(context)
+                                              .load(CoverHelper.buildGameTDBUrl(gameFile, "EN"))
+                                              .centerCrop()
+                                              .error(R.drawable.no_banner)
+                              )
```

Comment:
Let's say that a game only has cover art for EN. We go through this function, and the EN cover is downloaded. The next time this function is called, will it attempt to download the US cover and the other cover again, even though we already have the EN cover and it's unlikely that new cover variants have been added since then? If it does attempt to download the covers again, I think this can add up to a lot of network requests.